### ISSUE TITLE
Obnoxious vagrant warnings on darwin

### ISSUE THREAD

**Issue by nixy** [NixOS/nixpkgs#25640]
## Issue description
Whenever you call vagrant on darwin it produces some warnings regarding some gems that it appears to be using. It appears to work fine, but this is fairly obnoxious and makes it unpleasant to parse the output.

This is mostly a cosmetic/UX experience as such output is a little alarming. I wish I knew more about Ruby so I could fix it meself, but I figured making an issue for this was appropriate.

### Steps to reproduce
On darwin:

    nix-shell -p vagrant
    vagrant

You will see the following

```
Ignoring ffi-1.9.14 because its extensions are not built.  Try: gem pristine ffi --version 1.9.14
Ignoring nokogiri-1.6.7.1 because its extensions are not built.  Try: gem pristine nokogiri --version 1.6.7.1
Ignoring unf_ext-0.0.7.2 because its extensions are not built.  Try: gem pristine unf_ext --version 0.0.7.2
Ignoring wdm-0.1.1 because its extensions are not built.  Try: gem pristine wdm --version 0.1.1
...
```

## Technical details

* System: darwin
* Nix version: 1.11.9
* Nixpkgs version: 17.09pre107131.c882b967c1
* Sandboxing enabled: No

**Comment by bac** [comment]
I see similar:

```
vagrant up
Ignoring ffi-1.9.14 because its extensions are not built.  Try: gem pristine ffi --version 1.9.14
Ignoring nokogiri-1.6.7.1 because its extensions are not built.  Try: gem pristine nokogiri --version 1.6.7.1
Ignoring unf_ext-0.0.7.2 because its extensions are not built.  Try: gem pristine unf_ext --version 0.0.7.2
```

The suggested fix e.g. `gem pristine ffi --version 1.9.14` produces:

```
ERROR:  While executing gem ... (Gem::Exception)
    Failed to find gems ["ffi"] = 1.9.14
```

**Comment by xiaogangwen** [comment]
You may need to completely uninstall and reinstall Vagrant. See the link below.

http://keruuweb.com/vagrant-アップデートに伴いextensions-are-not-buildが発生/

**Comment by nixy** [comment]
While that might apply to normal installation of Vagrant, these errors occur no matter how many times you install/uninstall it through Nix.

I assume this has something to do with the way Nix is installing it, but like I said I'm not terribly familiar with Ruby or Vagrant.

**Comment by orivej** [comment]
Fixed in #30094
